### PR TITLE
JMAP: only send alerts for VEVENTs

### PR DESCRIPTION
**Pull request by brong**
This will make dogfood less horrible for people in the meanwhile!

[ not to be merged:  problem will be solved with an alternate strategy ]

### CODE REVIEW

**Comment by rsto** [comment]
Looks good to me. Should we reapply the VEVENT patch also on the jmap-calendars-01 branch?

File: imap/caldav_alarm.c [hunk 1]
```diff
@@ -581,6 +581,12 @@ static time_t process_alarms(const char *mboxname, uint32_t imap_uid,
                              const char *userid, icaltimezone *floatingtz,
                              icalcomponent *ical, time_t lastrun, time_t runtime)
 {
+    /* we don't send alarms for anything except VEVENTS */
+    icalcomponent *comp = icalcomponent_get_first_real_component(ical);
+    icalcomponent_kind kind = icalcomponent_isa(comp);
+    if (kind != ICAL_VEVENT_COMPONENT)
+        return 0;
+
     struct get_alarm_rock rock =
```
Comment (dilyanpalauzov) [inline imap/caldav_alarm.c:590]: This is commits bb88f7f37e5fd3c3 and 0bbbf12a5d32fcb4434874863 back and forth.

**Comment by rjbs** [comment]
Possible better fix:
* have the notification handling software (pusher) ignore alarms from non-VEVENT
* drop this commit

**Comment by rsto** [comment]
> Possible better fix:
> 
>     * have the notification handling software (pusher) ignore alarms from non-VEVENT

This would put unnecessary load on calalarmd to store and notify about alarms that the installation chose to ignore anyway. I just created https://github.com/cyrusimap/cyrus-imapd/pull/5015 which already prevents calalarmd from putting those alarms in the database in the first place.

**Comment by rsto** [comment]
This got replaced by https://github.com/cyrusimap/cyrus-imapd/pull/5015

**Pull request closed without merging**